kernel-modules/probe/collector_probe.c

```diff
@@ -275,9 +390,135 @@ static __always_inline int enter_probe(long id, struct sys_enter_args* ctx) {
   evt_type = sc_evt->enter_event_type;
   drop_flags = sc_evt->flags;
 
-  // To satisfy some verifier requiremnts in later parts of the falco plumbing/fillers,
-  // it is necessary to copy the context onto the stack.
-  memcpy(stack_ctx.args, _READ(ctx->args), sizeof(unsigned long) * NUM_SYS_ENTER_ARGS);
+  /* Syscall tracepoints follow their own format (=structure) of arguments.
+   * Copying all arguments, e.g. with
+   *   memcpy(stack_ctx.args, _READ(ctx->args), sizeof(unsigned long) * NUM_SYS_ENTER_ARGS);
+   * would copy beyond the format and results in an EACCESS triggered by
+   * format/structure offset checking in the perf event subsystem when installing
+   * the BPF program.
+   * Also the system exit fillers require access to the arguments to fill event
+   * data.  Note that the syscall tracepoints do not provide any argument data
+   * in the syscall exit format.  Of course, trying to access argument data in
+   * filler results in BPF verifier errors.
+   *
+   * To provide arguments to the syscall enter and exit fillers, extract the
+   * arguments from syscall specific formats/structures to an sys_enter_args
+   * structure and use a BPF map to make them available to the enter/exit
+   * fillers.
+   */
+  switch (id) /* maybe use sc_evt here..... */
+  {
+#ifdef __NR_accept
+    case __NR_accept: {
+      struct sys_enter_accept4_args* accept4_args = (struct sys_enter_accept4_args*)ctx;
+      stack_ctx.args[0] = (unsigned long)accept4_args->fd;
+      stack_ctx.args[1] = (unsigned long)accept4_args->upeer_sockaddr;
+      stack_ctx.args[2] = (unsigned long)accept4_args->upeer_len;
+      break;
+    }
+#endif
+    case __NR_accept4: {
+      struct sys_enter_accept4_args* accept4_args = (struct sys_enter_accept4_args*)ctx;
+      stack_ctx.args[0] = (unsigned long)accept4_args->fd;
+      stack_ctx.args[1] = (unsigned long)accept4_args->upeer_sockaddr;
+      stack_ctx.args[2] = (unsigned long)accept4_args->upeer_len;
+      stack_ctx.args[2] = (unsigned long)accept4_args->flags;
```

Comment:
self-review: This needs to be `stack_ctx.args[3]`.

@Molter73 @kcrane This might explain seeing container network related issues on the accepting/server side.

Reply:
Corrected now.